macosx/InfoGeneralViewController.mm

```diff
@@ -60,9 +62,16 @@ - (void)updateInfo
     Torrent* torrent = self.fTorrents[0];
 
     NSString* location = torrent.dataLocation;
+    NSString* lastKnownDataLocation = torrent.lastKnownDataLocation;
+
     self.fDataLocationField.stringValue = location ? location.stringByAbbreviatingWithTildeInPath : @"";
     self.fDataLocationField.toolTip = location ? location : @"";
 
+    self.fLastDataLabel.hidden = !!location;
+    self.fLastDataLocationField.hidden = !!location;
+    self.fLastDataLocationField.stringValue = !location ? lastKnownDataLocation.stringByAbbreviatingWithTildeInPath : @"";
```

Comment:
nit: Maybe it would be better to have here a normal `if/else` blocks?

Like:

```objc
if (location != nil)
{
    self.fLastDataLabel.hidden = YES;
    ...
}
else
{
    self.fLastDataLabel.hidden = NO;
    ...
}
```

Reply:
That would double the number of lines. I've made each line start with `location ?` now, so it's more readable in 939c060a63d22263f2f91eb00e91a942628137d2.